### PR TITLE
fix(fontupload): Allow user uploaded fonts and apply

### LINKED ISSUE #170: feat(add): Clicking OpenFolder does not open the Emojis folder
**Issue:**
Like title says Clicking OpenFolder does not open the Emojis folder and needs hooked up 

**Screen recording:**

https://github.com/Satellite-im/UplinkWeb/assets/93608357/3fc38973-2f4a-47cd-9d85-591b2b7cd243

### PR DESCRIPTION
**Pull request by Jekrimo**
### What this PR does 📖

-Adds save files specifically for font uploads. Creates short lived blob urls to handle files in indexdb and keeps track locally in svelte Stores.

### Which issue(s) this PR fixes 🔨

- Resolve #170 

### Special notes for reviewers 🗒️

Ticket was originally for uploading emoji but this involves some more discussion as pointed out by Shelly, uploaded emoji would only work for user that uploaded and would be broken when sent to another user, (unless the receiver has the same installed emoji)

Current functionality allows upload of specific types of files (we can add extra but these are the ones I found), .woff, .woff2, .otf, and .ttf

Some notes on current functionality: 

uploaded fonts should load in and set automatically on user upload

user uploads should persist across the app on navigation but also page refreshes

### Additional comments 🎤

Currently there is no error (toast) notifications if a user uploads a non-font file within font upload, (noted above in excepted file types) I am working on this currently

### CODE REVIEW

**Comment by luisecm** [comment]
@Jekrimo The automated tests failing are related to the change on the following UI locator. I went ahead and added a pr in testing repo to handle this change in tests: https://github.com/Satellite-im/automated-tests-web/pull/160

image
image
image

**Comment by Jekrimo** [comment]
Thanks @luisecm I will keep these in mind in the future for future pr's so I don't give you more work!

**Comment by InfamousVague** [comment]
This looks great, the only thing I'd say we should change is let's remove the theme code since that is introduced in another PR

File: src/routes/+layout.svelte [hunk 1]
```diff
@@ -171,24 +229,24 @@
...
-        await checkIfUserIsLogged($page.route.id)
-        await initializeLocale()
-    })
-
     let isLocaleSet = false
 
     $: if ($locale) {
         isLocaleSet = true
     }
+    // $: theme = UIStore.state.theme
```
Comment (stavares843) [inline src/routes/+layout.svelte:237]: is this on purpose?

File: src/routes/settings/preferences/+page.svelte [hunk 2]
```diff
@@ -98,44 +303,78 @@
...
+            options={$availableEmojiStore}
             alt
             on:change={v => {
                 UIStore.setEmojiFont(v.detail)
             }} />
-        <Button hook="button-emoji-font-open-folder" icon appearance={Appearance.Alt} tooltip={$_("generic.openFolder")}>
+        <Button
+            hook="button-emoji-font-open-folder"
+            on:click={async event => {
+                // let emojiUpload = event.dataTransfer?.files
```
Comment (stavares843) [inline src/routes/settings/preferences/+page.svelte:347]: do we need this?

Comment (Jekrimo) [reply src/routes/settings/preferences/+page.svelte:347]: Thank you for the catches Sara! I'mma delete them now!

**Comment by Jekrimo** [comment]
@InfamousVague Do you think we should hide the file upload buttons for these for now? 

image

I haven't checked out the identicon/theme needs for the app in terms of file type, limitations, and such,  but we had talked in the meeting about the emoji not working for the receiver and the issue that can cause having the glyphs come through as gobblygook for said reciever.

**Comment by luisecm** [comment]
Found one bug introduced

When you go to Settings Customizations for the first time it looks like this in your branch:

- Moon theme button appears as "clear button". Also "Default" is not showing in dropdown selected. After that, if you click it, button changes to moon theme
- Currently on dev, when going for the first time to screen, moon button appears and Default appears as dropdown value

image

**Comment by Jekrimo** [comment]
That was one billion percent my fault. I was trying to type our some stuff for themes but after I broke it I missed some stuff, my bad! I removed all the stuff that was messing it up and it should be good now! Thanks @luisecm !

**Comment by luisecm** [comment]
@Jekrimo all good now, thanks for fixing it!

**Comment by Jekrimo** [comment]
Added a couple extra checks today to weed out too large of files and unsupported files

**Review by InfamousVague — APPROVED** [review]

**Pull request merged**